src/components/HeadingInlineForm.js

```diff
@@ -19,87 +19,69 @@ const styles = {
   },
 };
 
-class HeadingInlineForm extends Component {
-  constructor(props) {
-    super(props);
+const HeadingInlineForm = ({ itemId, cancelFn, saveFn }) => {
+  // State variables from Redux store
+  const { smData } = useSelector((state) => state.structuralMetadata);
 
-    // To implement validation logic on begin and end times, we need to remove the current item
-    // from the stored data
-    this.tempSmData = undefined;
-  }
-
-  static propTypes = {
-    item: PropTypes.object,
-    cancelFn: PropTypes.func,
-    saveFn: PropTypes.func,
-  };
-
-  state = {
-    headingTitle: '',
-  };
-
-  componentDidMount() {
-    const { smData } = this.props;
+  const [headingTitle, setHeadingTitle] = useState('');
 
+  useEffect(() => {
     // Get a fresh copy of store data
-    this.tempSmData = cloneDeep(smData);
+    const tempSmData = cloneDeep(smData);
 
     // Load existing form values
-    this.setState(getExistingFormValues(this.props.item.id, this.tempSmData));
-  }
+    const formValues = getExistingFormValues(itemId, tempSmData);
+    setHeadingTitle(formValues.headingTitle);
+  }, [smData]);
 
-  formIsValid() {
-    return isTitleValid(this.state.headingTitle);
-  }
+  const formIsValid = () => {
+    return isTitleValid(headingTitle);
+  };
 
-  handleCancelClick = () => {
-    this.props.cancelFn();
+  const handleCancelClick = () => {
+    cancelFn();
   };
 
-  handleInputChange = (e) => {
-    this.setState({ [e.target.id]: e.target.value });
+  const handleInputChange = (e) => {
+    setHeadingTitle(e.target.value);
```

Comment:
I like how this is explicit about what is getting set (title) instead of a generic state change of whatever the event target passed.